plugins/generic/externalFeed/controllers/grid/ExternalFeedGridHandler.inc.php

```diff
@@ -0,0 +1,244 @@
+	 */
+	static public function setPlugin($plugin) {
+		self::$plugin = $plugin;
+	}
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		parent::__construct();
+		$this->addRoleAssignment(
+			array(ROLE_ID_MANAGER),
+			array('index', 'fetchGrid', 'fetchRow', 'addExternalFeed', 'editExternalFeed', 'updateExternalFeed', 'delete')
+		);
+	}
+
+	//
+	// Overridden template methods
+	//
+	/**
+	 * @copydoc Gridhandler::initialize()
+	 */
+	public function initialize($request, $args = null) {
+		parent::initialize($request);
+		$context = $request->getContext();
+		
+		// Set the grid details.
+		$this->setTitle('plugins.generic.externalFeed.displayName');
+		$this->setEmptyRowText('plugins.generic.externalFeed.manager.noneCreated');
+		
+		// Get the pages and add the data to the grid
+		self::$plugin->import('classes.ExternalFeed');
+		$externalFeedDao = DAORegistry::getDAO('ExternalFeedDAO');
+		$feeds = $externalFeedDao->getExternalFeedsByJournalId($context->getId());
+		$this->setGridDataElements($feeds);
+		
+		// Add grid-level actions
+		$router = $request->getRouter();
+		import('lib.pkp.classes.linkAction.request.AjaxModal');
+		$this->addAction(
+			new LinkAction(
+				'addExternalFeed',
+				new AjaxModal(
+						$router->url($request, null, null, 'addExternalFeed'),
+						__('plugins.generic.externalFeed.manager.create'),
+						'modal_add_item'
+						),
+				__('plugins.generic.externalFeed.manager.create'),
+				'add_item'
+			)
+		);
+		
+		// Columns
+		$cellProvider = new ExternalFeedGridCellProvider();
+		$this->addColumn(new GridColumn(
+			'title',
+			'plugins.generic.externalFeed.manager.title',
+			null,
+			'controllers/grid/gridCell.tpl',
+			$cellProvider
+		));
+		$this->addColumn(new GridColumn(
+			'homepage',
+			'plugins.generic.externalFeed.manager.displayHomepage',
+			null,
+			'controllers/grid/common/cell/selectStatusCell.tpl',
+			$cellProvider
+		));
+		$this->addColumn(new GridColumn(
+			'displayBlockAll',
+			'plugins.generic.externalFeed.manager.displayBlockAll',
+			null,
+			'controllers/grid/common/cell/selectStatusCell.tpl', 
+			$cellProvider
+		));
+		$this->addColumn(new GridColumn(
+			'displayBlockHomepage',
+			'plugins.generic.externalFeed.manager.displayBlockHomepage',
+			null,
+			'controllers/grid/common/cell/selectStatusCell.tpl',
+			$cellProvider
+		));
+		
+	}
+
+	/**
+	 * @copydoc GridHandler::initFeatures()
+	 */
+	function initFeatures($request, $args) {
+		import('lib.pkp.classes.controllers.grid.feature.OrderGridItemsFeature');
+		return array(new OrderGridItemsFeature());
+	}
+
+	/**
+	 * @copydoc GridHandler::getDataElementSequence()
+	 */
+	function getDataElementSequence($row) {
+		return $row->getSequence();
+	}
+
+	/**
+	 * @copydoc GridHandler::setDataElementSequence()
+	 */
+	function setDataElementSequence($request, $rowId, $gridDataElement, $newSequence) {
+		$externalFeedId = $rowId;
+		$journal = $request->getJournal();

```

Comment:
I suspect this could be made OMP-compatible by changing a bit of OJS-specific terminology (e.g. `$journal = $request->getJournal`) into app-agnostic terminology (e.g. `$context = $request->getContext()`).